### PR TITLE
[Nokia-IXER7250E] Modify the Supervisor device data to keep SFM alive while ungraceful reboot

### PR DESCRIPTION
**Pull request by mlok-nokia**
#### Why I did it
On Nokia-IXR7250E chassis,  Fabrics Cards need to be up and running to keep the iBGP traffic intact while Supervisor is reboot ungracefully.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Modify the Supervsor platform_ndk.json to enable ungraceful_reboot_keep_sfm_alive flag  to notify the device-manager not to powercycle the SFMs.  And set the ungraceful_reboot_log_interval to 10 seconds to periodically notify users by logging that that system is ungraceful reboot,  need to reboot the whole chassis to recover to a good state.  Also set ungraceful_reboot_sfm_online to "no" to avoid the related swss/syncd to reinitialize the SFMs which trigger the SFMs down/up.

#### How to verify it
1) On supervisor, execute the "sudo /sbin/reboot"
2) after Supervisor is up,  check the syslog, the following message will be logged in every 10 seconds
```
Dec  8 19:04:21.428535 Supervisor ERR sr_device_mgr: debug|16648|16648|00244|E: common    device_mgr.cc:1958       CpmLogWarmRestartMsg  Supervisor card has been ungracefully rebooted. Require to 'sudo reboot' the whole chassis to recover to a good state

```
3)  docker -ps  to verify swss/syncd should not be up.
```
admin@ixre-cpm-chassis14:~$ docker ps
CONTAINER ID   IMAGE                                COMMAND                  CREATED      STATUS         PORTS     NAMES
18824d15c6c3   docker-snmp:latest                   "/usr/local/bin/supe\u2026"   3 days ago   Up 3 minutes             snmp
b84500428805   docker-sonic-telemetry:latest        "/usr/local/bin/supe\u2026"   3 days ago   Up 3 minutes             telemetry
08909b4b0571   docker-sonic-mgmt-framework:latest   "/usr/local/bin/supe\u2026"   3 days ago   Up 3 minutes             mgmt-framework
fdf151c7fde0   docker-lldp:latest                   "/usr/bin/docker-lld\u2026"   3 days ago   Up 5 minutes             lldp
219d3eafa864   docker-platform-monitor:latest       "/usr/bin/docker_ini\u2026"   3 days ago   Up 5 minutes             pmon
e6ca3c20a3e7   docker-router-advertiser:latest      "/usr/bin/docker-ini\u2026"   3 days ago   Up 5 minutes             radv
0adca1fd2cdb   docker-database:latest               "/usr/local/bin/dock\u2026"   3 days ago   Up 5 minutes             database13
dbd33f387b24   docker-database:latest               "/usr/local/bin/dock\u2026"   3 days ago   Up 5 minutes             database10
6e30004405c5   docker-database:latest               "/usr/local/bin/dock\u2026"   3 days ago   Up 5 minutes             database3
61dfda55f836   docker-database:latest               "/usr/local/bin/dock\u2026"   3 days ago   Up 5 minutes             database2
ae462ee9cbdf   docker-database:latest               "/usr/local/bin/dock\u2026"   3 days ago   Up 5 minutes             database4
7c8303c14b12   docker-database:latest               "/usr/local/bin/dock\u2026"   3 days ago   Up 5 minutes             database8
2457f156ad8c   docker-database:latest               "/usr/local/bin/dock\u2026"   3 days ago   Up 5 minutes             database7
014acd8cbecf   docker-database:latest               "/usr/local/bin/dock\u2026"   3 days ago   Up 5 minutes             database1
517a520718ca   docker-database:latest               "/usr/local/bin/dock\u2026"   3 days ago   Up 5 minutes             database6
8c8f14b1b006   docker-database:latest               "/usr/local/bin/dock\u2026"   3 days ago   Up 5 minutes             database12
a12d42f1e6fd   docker-database:latest               "/usr/local/bin/dock\u2026"   3 days ago   Up 5 minutes             database9
aceb87084a95   docker-database:latest               "/usr/local/bin/dock\u2026"   3 days ago   Up 5 minutes             database5
6b90bd8913f2   docker-database:latest               "/usr/local/bin/dock\u2026"   3 days ago   Up 5 minutes             database15
07490dc19eb2   docker-database:latest               "/usr/local/bin/dock\u2026"   3 days ago   Up 5 minutes             database14
dbce0be9146f   docker-database:latest               "/usr/local/bin/dock\u2026"   3 days ago   Up 5 minutes             database11
72f05cdd7030   docker-database:latest               "/usr/local/bin/dock\u2026"   3 days ago   Up 5 minutes             database0
dcfa638f3c47   docker-database:latest               "/usr/local/bin/dock\u2026"   3 days ago   Up 5 minutes             database
0294d839df01   docker-database:latest               "/usr/local/bin/dock\u2026"   3 days ago   Up 5 minutes             database-chassis

```

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [x] 202205
- [x] 202211
- [x] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by mlok-nokia** [comment]
@judyjoseph  This is the PR which enable to keep SFMs alive while Supervisor is ungraceful reboot.

File: device/nokia/x86_64-nokia_ixr7250e_sup-r0/platform_ndk.json [hunk 1]
```diff
@@ -31,6 +31,38 @@
...
+    {
+      "key": "ungraceful_reboot_keep_sfm_alive",
+      "stringval": "yes"
+    },
+    {
+      "key": "ungraceful_reboot_log_interval",
+      "intval": 10
+    },
+    {
+      "key": "ungraceful_reboot_sfm_online",
```
Comment (judyjoseph) [inline device/nokia/x86_64-nokia_ixr7250e_sup-r0/platform_ndk.json:44]: There are two similar attributes to not reset SFM on SUP restart -  ungraceful_reboot_sfm_online and ungraceful_reboot_keep_sfm_alive. Are they both for different purposes ?

Comment (mlok-nokia) [reply device/nokia/x86_64-nokia_ixr7250e_sup-r0/platform_ndk.json:44]: > There are two similar attributes to not reset SFM on SUP restart - ungraceful_reboot_sfm_online and ungraceful_reboot_keep_sfm_alive. Are they both for different purposes ?

Yes.   ungraceful_reboot_keep_sfm_alive is for not to powercycle the SFMs when it is a ungraceful reboot.   ungraceful_reboot_sfm_online is for allowing NDK to report SFM "online" status to PMON API query which will bring up the related  swss/syncd   
"ungraceful_reboot_keep_sfm_alive" enable/disable the ungraceful reboot support while "ungraceful_reboot_sfm_online"  bring up or keep down the related swss/syncd when Supervisor is ungraceful reboot.

**Comment by mlok-nokia** [comment]
No longer applicable due the new behaviour requirement of voq-bgp-tsa/tsb control

**Pull request closed without merging**